src/coreclr/src/vm/eventpipesession.cpp

```diff
@@ -482,12 +567,32 @@ void EventPipeSession::SuspendWriteEvent()
         THROWS;
         GC_TRIGGERS;
         MODE_PREEMPTIVE;
+        // Need to disable the session before calling this method
+        PRECONDITION(HasOnlyRundownProviders());
```

Comment:
This is true, though it feels like an odd thing to assert. Asserting that the s_pSession array slot is null seems like a more direct way to validate that the session will not be writing any more events?

Reply:
I just got tunnel visioned. I originally had !IsEnabled(), but the rundown provider was sometimes there so I created this instead of rethinking the approach